Jenkinsfile

```diff
@@ -910,13 +910,30 @@ def pullContainerImages(String mavenArgs) {
 	if ( containerImageRefs.length == 0 ) {
 		return
 	}
-	docker.withRegistry('https://index.docker.io/v1/', 'hibernateci.hub.docker.com') {
-		// Cannot use a foreach loop because then Jenkins wants to serialize the iterator,
-		// and obviously the iterator is not serializable.
-		for (int i = 0; i < containerImageRefs.length; i++) {
-			containerImageRef = containerImageRefs[i]
-			docker.image( containerImageRef ).pull()
+	String[] dockerRegistries = ['https://docker.elastic.co/v1/', 'https://index.docker.io/v1/']
+	for (i in 0..<dockerRegistries.length) {
+		if (containerImageRefs.length == 0) {
+			break
 		}
+		List<String> failedImageRefs = []
+		docker.withRegistry(dockerRegistries[i], 'hibernateci.hub.docker.com') {
```

Comment:
`hibernateci.hub.docker.com` points to credentials that will only work on dockerhub, so I'm not sure what you're going for here... ?

Essentially you only need `withRegistry()` when you need authentication, which we don't need for docker.elastic.co.

FWIW, when you don't need authentication, you also don't really need this pre-build pulling. We could simply filter out all images that obviously are not on dockerhub (because they start with `docker.elastic.co/`, ...) and have them pulled during the build.